src/axom/inlet/Container.hpp

```diff
@@ -1231,6 +1231,34 @@ class Container : public Verifiable<Container>
   template <typename Func>
   void forEachCollectionElement(Func&& func) const;
 
+  /*!
+   *****************************************************************************
+   * \brief Applies a provided function to nested elements of the calling table
+   * and stores the result in a range
+   * 
+   * \pre The function must accept two arguments of type Table& and
+   * const std::string&, respectively.  
+   * 
+   * This function will pass to the provided function the nested table
+   * as the first argument and the path of the nested element in the input file
+   * as the second argument, when applicable.
+   * 
+   * \param [out] output An iterator to the beginning of the output range
```

Comment:
Minor / convention -- is it weird for the first parameter to be an output parameter, while the others are input parameters? Would users expect the output parameters to be at the end?

Reply:
Good question - STL algorithms like [std::transform](https://en.cppreference.com/w/cpp/algorithm/transform) typically always keep the callable at the end for readability when passing a lambda as an argument, and that's what I've opted to go with there.  I don't have a preference regarding the order of the first two, though, so I can flip those if you'd prefer.

Reply:
That makes sense.  I'm ok with it as is.